### PR TITLE
exclude onsiteSearch clicks from event22

### PR DESCRIPTION
**Pull request by nickyanalytics**

### CODE REVIEW

File: tests/doplugins/doplugins_article_view_type.test.js [hunk 1]
```diff
@@ -454,6 +454,21 @@ describe('articleViewType()', () => {
...
+            [false, 'article : 245145230'],
+        ];
+        it.each(testData)('should return true for previousPage: "search : 245145230" ', function (exected, _ppvPreviousPage) {
+            Object.assign(sObject, { _ppvPreviousPage });
+
+            const result = s._articleViewTypeObj.isFromOnsiteSearch();
+            expect(result).toBe(exected);
+        });
+    });
+
```
Comment (mehakraza) [inline tests/doplugins/doplugins_article_view_type.test.js:471]: escribe('isFromOnsiteSearch', () => {
    const testData = [
        [false, undefined],
        [false, ''],
        [true, 'search : 245145230'],
        [false, 'article : 245145230'],
    ];
    
    it.each(testData)('should return **%s** for previousPage: **"%s"**', function (**expected**, _ppvPreviousPage) {
        Object.assign(sObject, { _ppvPreviousPage });

        const result = s._articleViewTypeObj.isFromOnsiteSearch();
        expect(result).toBe(**expected**);
    });
});

File: tests/doplugins/doplugins_article_view_type.test.js [hunk 2]
```diff
@@ -454,6 +454,21 @@ describe('articleViewType()', () => {
...
+            [false, undefined],
+            [false, ''],
+            [true, 'search : 245145230'],
+            [false, 'article : 245145230'],
+        ];
+        it.each(testData)('should return true for previousPage: "search : 245145230" ', function (exected, _ppvPreviousPage) {
+            Object.assign(sObject, { _ppvPreviousPage });
+
+            const result = s._articleViewTypeObj.isFromOnsiteSearch();
+            expect(result).toBe(exected);
```
Comment (mehakraza) [inline tests/doplugins/doplugins_article_view_type.test.js:468]: expect(result).toBe(expected);

File: tests/doplugins/doplugins_article_view_type.test.js [hunk 3]
```diff
@@ -475,19 +490,22 @@ describe('articleViewType()', () => {
...
+            isFromOnsiteSearchMock = jest.spyOn(s._articleViewTypeObj, 'isFromOnsiteSearch').mockReturnValue(false);
         });
 
-        it('should return event22 if referrer is a home page and page was navigated and it was no selfRedirect', function () {
+        it('should return event22 if referrer is a home page and page was navigated and it was no selfRedirect and not via OnsiteSearch', function () {
             const anyReferrer = 'http://www.any-domain.de';
             isFromHomeMock.mockReturnValue(true);
             isNavigatedMock.mockReturnValue(true);
             isSelfRedirectMock.mockReturnValue(false);
+            isFromOnsiteSearchMock.mockReturnValue(false);
```
Comment (mehakraza) [inline tests/doplugins/doplugins_article_view_type.test.js:508]: It is not needed to return 'false' value as this is already done before each test in line 500.

File: tests/doplugins/doplugins_article_view_type.test.js (hunk 2 above)
Comment (nickyanalytics) [reply tests/doplugins/doplugins_article_view_type.test.js:468]: I vote for exected :-)

File: tests/doplugins/doplugins_article_view_type.test.js (hunk 3 above)
Comment (nickyanalytics) [reply tests/doplugins/doplugins_article_view_type.test.js]: Thank you :-)

**Pull request merged**